sdkproject/Assets/Mapbox/Unity/MeshGeneration/Modifiers/MeshModifiers/ReplaceFeatureModifier.cs

```diff
@@ -48,42 +57,72 @@ public override void SetProperties(ModifierProperties properties)
 			_options = (SpawnPrefabOptions)properties;
 		}
 
+		public override void FeaturePreProcess(VectorFeatureUnity feature)
+		{
+			int index = -1;
+			foreach (var point in _prefabLocations)
+			{
+				try
+				{
+					index++;
+					var coord = Conversions.StringToLatLon(point);
+					if (feature.ContainsLatLon(coord))
+					{
+						if (feature.Data.Id != 0)
+						{
+							if (_featureId[index] == null)
+							{
+								_featureId[index] = new List<string>();
+							}
+							_tempFeatureId = feature.Data.Id.ToString();
+							_featureId[index].Add(_tempFeatureId.Substring(0, _tempFeatureId.Length - 3));
```

Comment:
Why `- 3`?

Reply:
that's because parts of the same building come with their IDs that are common until the last 3 digits

Reply:
it's a hacky solution to address some issues with our data